services-custom/s3-transfer-manager/src/main/java/software/amazon/awssdk/transfer/s3/S3TransferManager.java

```diff
@@ -276,6 +276,54 @@ default DirectoryUpload uploadDirectory(Consumer<UploadDirectoryRequest.Builder>
         return uploadDirectory(UploadDirectoryRequest.builder().applyMutation(requestBuilder).build());
     }
 
+    /**
+     * Download all objects under a specific prefix and bucket to the provided directory. By default, all objects in the entire
+     * bucket will be downloaded.
```

Comment:
More documentation such as whether we require the directory to pre-exist or not will be added as we start to work on implementation.